docs/mobile-apps/automated-testing/appium/real-devices.md

```diff
@@ -2,190 +2,209 @@
---form 'name="iOS.RealDevice.SauceLabs.Mobile.Sample.app.2.7.1.ipa"'
-```
+### Excluding the `browserName`
 
-US Data Center (Windows)
+When testing a native mobile app, no browser is accessed, so if you are re-using the capabilities from your mobile or desktop browser tests, you can omit the `browserName` capability or or set the value as an empty string. This is an important exclusion because if values are set for _both_ `app` and `browserName`, Sauce Labs defaults to the `browserName`. Similarly, if neither capability is specified, Sauce Labs defaults to the `browserName` settings of Safari for iOS and Chrome for Android.
 
-```zh
-> curl -u "%SAUCE_USERNAME%:%SAUCE_ACCESS_KEY% -X POST -w "%{http_code}\n" \
--H "Content-Type: application/octet-stream" \
-"https://saucelabs.com/rest/v1/storage/%SAUCE_USERNAME%/Android.SauceLabs.Mobile.Sample.app.x.x.x.apk?overwrite=true" \
---data-binary @\path\to\Android.SauceLabs.Mobile.Sample.app.x.x.x.apk
+```js
+caps.setCapability("browserName", "");
 ```
 
-</TabItem>
-<TabItem value="EU Data Center">
+### Selecting Your Test Device
 
-EU Data Center (macOS/Linux)
+Testing on real devices requires you to specify a device on which you plan to test your app. You can do this by specifying either:
 
-```curl
-curl -u "$SAUCE_USERNAME:$SAUCE_ACCESS_KEY" --location \
---request POST 'https://api.eu-central-1.saucelabs.com/v1/storage/upload' \
---form 'payload=@"g16K4P8IX/iOS.RealDevice.SauceLabs.Mobile.Sample.app.2.7.1.ipa"' \
---form 'name="iOS.RealDevice.SauceLabs.Mobile.Sample.app.2.7.1.ipa"'
-```
+* A specific device from a Sauce Labs device pool (public or private) by its ID (static allocation)
+* A set of device attributes to use as criteria for selecting any available matching device (dynamic allocation)
+
+#### Static Device Allocation
 
-EU Data Center (Windows)
+_Static Allocation_ allows you to specify a known device by its unique ID. This can be beneficial if, for examplle, you are testing features only available on a very specific device setup. However, what you gain in precision may be offset by the time it takes for a specific device to become available, especially if your tests do not require that level of precision. If you do require a specific device, you should always configrm the device's availability before launching your tests.
 
-```bash
-> curl -u "%SAUCE_USERNAME%:%SAUCE_ACCESS_KEY%" -X POST -w "%{http_code}\n" \
--H "Content-Type: application/octet-stream" \
-"https://eu-central-1.saucelabs.com/rest/v1/storage/%SAUCE_USERNAME%/Android.SauceLabs.Mobile.Sample.app.x.x.x.apk?overwrite=true" \
---data-binary @\path\to\Android.SauceLabs.Mobile.Sample.app.x.x.x.apk
+```js
+caps.setCapability("deviceName", "HTC_One_M8_real");
 ```
 
-</TabItem>
-</Tabs>
+:::note
+When you specify a device by it's ID, the `platformName` and `platformVersion` capabilities are populated automatically, so may be omitted in your test configuration. If included, they are ignored.
```

Comment:
This is not correct, `platformName` IS mandatory, `platformVersion` is optional

Reply:
No, I'm correct 😉 , here's an example. When you run this

```js
  {
    deviceName: 'iPhone_XS_ws',
    platformName: 'Android',
  },
```

You will get this error

```log
Request failed with status 500 due to There is no device that matches your criteria. DesiredCapabilities: {"desiredCapabilities":{"testobject_device":"iPhone_XS_ws","testobject_platform_name":"Android","platformName":"Android","uiautomator2ServerInstallTimeout":"60000","deviceName":"iPhone_XS_ws"}}
```

Running this, thus without the `platformName`

```js
  {
    deviceName: 'iPhone_XS_ws',
  },
```
 also fails

```log
[0-0] 2021-11-18T06:23:17.529Z DEBUG webdriver: request failed due to missing body
[0-0] 2021-11-18T06:23:17.529Z WARN webdriver: Request failed with status 400 due to Invalid device name specified: iPhone_XS_ws
```

Reply:
Ha! I should have known better. Note removed.